src/parsers/timezone.rs

```diff
@@ -77,6 +77,10 @@ pub(crate) fn parse_offset(chars: &mut Peekable<Chars<'_>>) -> TemporalResult<i1
     // First offset portion
     let hours = parse_digit_pair(chars)?;
 
+    if !(0..24).contains(&hours) {
```

Comment:
This was implemented in `temporal_rs` because there is not really any exported `OffsetParser` from `ixdtf`. I was hoping this could be pretty minimal, which is greedy. I do think it'd be best to try and upstream this to `ixdtf` if possible, especially to unify the parsing logic. In the meantime, this does fix the bugs related to it.

Reply:
Can you open an issue for it then? Just to keep track of the things that we need to upstream.

Reply:
Opened #226